lang/py/src/avro/schema.py

```diff
@@ -405,6 +420,106 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale):
+    LogicalSchema.__init__(self, decimal_type, 'decimal')
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def validate(self, validation_props=None):
```

Comment:
Is this method part of the public API? If so, we should probably make it work even when no `validation_props` are passed in: it's not clear to the user what they should pass in or even why they need to pass something in. (Since it's already called inside the constructor, it might be simpler to just make it private.)